### PR TITLE
Enable Solhint

### PR DESCRIPTION
**Pull request by RensR**
- Add most folders to ignore file
- Fixed `shared` folder solhint issues
- Add solhint to CICD

Each project should remove their own folder from the ignore list and fix any issues that come up.

### CODE REVIEW

File: CODEOWNERS
Comment (makramkd) [inline CODEOWNERS:1]: GitHub is saying this file has errors, but I don't think it's from this PR.

File: CODEOWNERS [hunk 1]
```diff
@@ -67,12 +67,12 @@ core/scripts/gateway @bolekk @pinebit
 /operator-ui/ @DeividasK @jkongie
 
 # Contracts
-/contracts/ @se3000 @connorwstein
+/contracts/ @se3000 @connorwstein @RensR
```
Comment (makramkd) [inline CODEOWNERS:70]: Will we have more granular code owners on this?

File: contracts/src/v0.8/automation/v2_0/KeeperRegistry2_0.sol [hunk 2]
```diff
@@ -5,7 +5,7 @@ import "../../vendor/openzeppelin-solidity/v4.7.3/contracts/proxy/Proxy.sol";
 import "../../vendor/openzeppelin-solidity/v4.7.3/contracts/utils/structs/EnumerableSet.sol";
 import "../../vendor/openzeppelin-solidity/v4.7.3/contracts/utils/Address.sol";
 import "./KeeperRegistryBase2_0.sol";
-import {AutomationRegistryExecutableInterface, UpkeepInfo} from "../interfaces/v2_0/AutomationRegistryInterface2_0.sol";
+import {AutomationRegistryExecutableInterface, UpkeepInfo, State, OnchainConfig, UpkeepFailureReason} from "../interfaces/v2_0/AutomationRegistryInterface2_0.sol";
```
Comment (makramkd) [inline contracts/src/v0.8/automation/v2_0/KeeperRegistry2_0.sol:8]: What was the warning here? Implicit import of `State`?

**Comment by makramkd** [comment]
LGTM overall, just minor comments from above about the CODEOWNERS file. Not sure if it's broken.

File: CODEOWNERS (hunk 1 above)
Comment (RensR) [reply CODEOWNERS:70]: Anything below that is more specific overwrites this. I see VRF is missing, but most other teams have their team as owner. Having this top level simply means that anything that falls between the cracks is caught by the people here (e.g. shared folder, which is sorta managed by me)

File: contracts/src/v0.8/automation/v2_0/KeeperRegistry2_0.sol (hunk 2 above)
Comment (RensR) [reply contracts/src/v0.8/automation/v2_0/KeeperRegistry2_0.sol:8]: I think due to other imports being made more explicit, it no longer imported these via others so had to add them explicitly here

**Review by makramkd — APPROVED** [review]

**Pull request merged**